rep-0145.rst

```diff
@@ -0,0 +1,207 @@
+
+* If the device does not have an absolute gravity reference (accelerometer), the world frame is not aligned to any external reference and instead aligned relative to the power-on position of the sensor.
+
+* The `frame_id` for all message types published by an IMU represents the body frame - the default frame ID for IMUs is `imu_link`. In compliance with REP 0103 [1]_, and as a hint to integrators, the default frame name for IMUs that use an NED world reference should be `imu_link_ned`.
+
+  - The configuration of the body frame relative to other frames (e.g. `base_link`) represents the mounting position of the IMU [2]_.
+
+  - The world frame orientation depends on the IMU device, and is not explicitly defined in the transform tree.
+
+* The device has an associated *neutral orientation*, defined as the orientation of the device where the body and the world frame align [3]_.
+
+Data Reporting
+--------------
+
+* To maintain interoperability with ROS conventions, all frames must be right handed. If any data is reported left handed the driver must convert it to right handed by inverting the `y` axis.
+
+* All data from a sensor should be published with respect to a single consistent body frame. If any data is reported in an inconsistent frame of reference relative to the other data, the driver must transform it into the body frame before publishing.
+
+* Otherwise, all data should be published by the driver as it is reported by the device. Any subsequent modifications to the data (e.g. filtering, transformations) should be delegated to a downstream consumer of the data [3]_.
+
+* A prominent note should be made in the driver documentation regarding any internal data manipulation that does not comply with the requirements in this document.
+
+Raw Data
+''''''''
+
+* Accelerometers
+
+  - The accelerometers report linear acceleration data in the body frame of the device. This data is output from the driver as a 3D vector, with the components representing the deflection of the internal accelerometers.
+
+  - When the device is at rest, the vector will represent the deflection solely due to gravity, and will always point 'up' away from the earth's gravitational center. This data must be in m/s^2.
+
+* Gyroscopes
+
+  - The gyroscopes report rotational velocity data in the body frame of the device. This data is output from the driver as a 3D vector, with the components representing the instantaneous velocity around each equivalent axis of the body frame.
+
+  - The rotational velocity is right handed with respect to the bode axes, and independent of the orientation of the device. This data must be in rad/s.
+
+* Magnetometers
+
+  - The magnetometers report magnetic field strength in the body frame of the device. This data is output from the driver as a 3D vector, with the components representing magnetic field strength in each direction. This data must be in Tesla.
```

Comment:
[Magnetic deviation](http://en.wikipedia.org/wiki/Magnetic_deviation) is another source of error caused my magnetized metal parts on the robot itself. Deviation errors can usually be calibrated to zero in the sensor directly or by applying a constant offset to the sensor readings in the body frame. It should be specified whether the data on topics is already calibrated (field `y` is 0 if the x-axis points to magnetic north) or not (raw measurements).

Would it be convenient to publish raw magnetic field strength on topic `imu/mag_raw` and the corrected magnetic field on topic `imu/mag`? A driver could provide both topics or only one of them. If a user wants to ignore deviation he could use a remapping to forward uncalibrated data to a node which expects a calibrated magnetic field.